test/e2e/framework/util.go

```diff
@@ -2162,6 +2162,10 @@ func (f *Framework) MatchContainerOutput(
 	podClient := f.PodClient()
 	ns := f.Namespace.Name
 
+	defer func() {
+		By("delete the pod")
+		podClient.Delete(pod.Name, nil)
```

Comment:
@ixdy : we don't really need the pods to be alive AFAICT from looking at the code paths